akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/ClusterShardingSpec.scala

```diff
@@ -261,12 +277,68 @@ class ClusterShardingSpec extends TestKit("ClusterShardingSpec", ClusterSharding
       val charlieRef = sharding.entityRefFor(typeKey, "charlie")
 
       val reply1 = bobRef ? WhoAreYou // TODO document that WhoAreYou(_) would not work
-      reply1.futureValue should ===("I'm bob")
+      reply1.futureValue should startWith("I'm bob")
 
       val reply2 = charlieRef ask WhoAreYou
-      reply2.futureValue should ===("I'm charlie")
+      reply2.futureValue should startWith("I'm charlie")
 
       bobRef ! StopPlz()
     }
+
+    "handle untyped StartEntity message" in {
+      // it is normally using envolopes, but the untyped StartEntity message can be sent internally,
+      // e.g. for remember entities
+
+      val totalCountBefore = totalEntityCount1()
+
+      val p = TestProbe[Any]()
+      shardingRef1.toUntyped.tell(akka.cluster.sharding.ShardRegion.StartEntity("startEntity-1"), p.ref.toUntyped)
+      p.expectMsgType[akka.cluster.sharding.ShardRegion.StartEntityAck]
+
+      eventually {
+        val totalCountAfter = totalEntityCount1()
+        totalCountAfter should ===(totalCountBefore + 1)
+      }
+    }
+
+    "handle typed StartEntity message" in {
+      val totalCountBefore = totalEntityCount1()
+      println(s"# totalCountBefore $totalCountBefore") // FIXME
+
+      shardingRef1 ! StartEntity("startEntity-2")
+
+      eventually {
+        val totalCountAfter = totalEntityCount1()
+        println(s"# totalCountAfter $totalCountAfter") // FIXME
+        totalCountAfter should ===(totalCountBefore + 1)
+      }
+    }
```

Comment:
Hmmm, but how do you do start-entity when you don't use envelopes? Did I miss that?

Reply:
ClusterSharding itself will send that message for remember entities. That is the original purpose of `StartEntity`.